src/arithmetic/rewrite_simplify.cc

```diff
@@ -1757,6 +1757,20 @@ Mutate_(const Variable* op, const Expr& self) {
   return self;
 }
 
+Expr RewriteSimplifier::Impl::
+Mutate_(const Cast* op, const Expr& self) {
```

Comment:
Let us just add cast between (uint/float/int here), ignore vector types for now as they are not as common.